ds3-sdk/src/test/java/com/spectralogic/ds3client/metadata/PosixMetadataRestore_Test.java

```diff
@@ -0,0 +1,131 @@
+import static java.nio.file.LinkOption.NOFOLLOW_LINKS;
+
+@RunWith(JUnit4.class)
+public class PosixMetadataRestore_Test {
+    private final File file  = new File(getClass().getClassLoader().getResource("LoremIpsum.txt").getFile());
+    private final String localOS = MetaDataUtil.getOS();
+
+
+
+
+    @Test
+    public  void restoreFileTimes_Test() throws Exception{
+        final BasicHeader basicHeader[] = new BasicHeader[3];
+        final BasicFileAttributes attr = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+        basicHeader[0] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_CREATION_TIME,String.valueOf(attr.creationTime().toMillis()));
+        basicHeader[1] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_ACCESS_TIME,String.valueOf(attr.lastAccessTime().toMillis()));
+        basicHeader[2] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_LAST_MODIFIED_TIME,String.valueOf(attr.lastModifiedTime().toMillis()));
+        final  Metadata metadata = genMetadata(basicHeader);
+        final PosixMetadataRestore posixMetadataRestore = new PosixMetadataRestore(metadata,file.getPath(),MetaDataUtil.getOS(),Mockito.mock(MetadataRestoreListener.class));
+        posixMetadataRestore.restoreFileTimes();
+        final BasicFileAttributes fileAttributes = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+        Assert.assertEquals(String.valueOf(fileAttributes.creationTime().toMillis()),String.valueOf(basicHeader[0].getValue()));
+        Assert.assertEquals(String.valueOf(fileAttributes.lastModifiedTime().toMillis()),String.valueOf(basicHeader[2].getValue()));
+    }
+
+
+      @Test
+      public void restoreUserAndOwner() throws IOException{
+          final int uid = (int) Files.getAttribute(file.toPath(), "unix:uid", NOFOLLOW_LINKS);
+          final int gid = (int) Files.getAttribute(file.toPath(), "unix:gid", NOFOLLOW_LINKS);
+          final BasicHeader basicHeader[] = new BasicHeader[2];
+          basicHeader[0] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_UID,String.valueOf(uid));
+          basicHeader[1] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_GID,String.valueOf(gid));
+          final  Metadata metadata = genMetadata(basicHeader);
+          final PosixMetadataRestore posixMetadataRestore = new PosixMetadataRestore(metadata,file.getPath(),MetaDataUtil.getOS(),Mockito.mock(MetadataRestoreListener.class));
+          posixMetadataRestore.restoreUserAndOwner();
+          Assert.assertEquals(String.valueOf((int) Files.getAttribute(file.toPath(), "unix:uid", NOFOLLOW_LINKS)),basicHeader[0].getValue());
+          Assert.assertEquals(String.valueOf((int) Files.getAttribute(file.toPath(), "unix:gid", NOFOLLOW_LINKS)),basicHeader[1].getValue());
+      }
+
+
+      @Test
+      public void restorePermissions() throws  Exception{
```

Comment:
This test fails on windows

java.lang.UnsupportedOperationException
	at sun.nio.fs.WindowsFileSystemProvider.readAttributes(WindowsFileSystemProvider.java:192)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at com.spectralogic.ds3client.metadata.PosixMetadataRestore_Test.restorePermissions(PosixMetadataRestore_Test.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)